app/recordtransfer/views/transfer.py

```diff
@@ -157,74 +155,77 @@ def current_step(self) -> TransferStep:
         if not self.in_progress_submission:
-            LOGGER.error(
-                "Expected at least 1 saved transfer for user %s and ID %s, found 0",
-                request.user,
-                self.in_progress_uuid,
-            )
             return redirect("recordtransfer:transfer")
 
-        return None
+        return super().dispatch(request, *args, **kwargs)
 
-    def get(self, request, *args, **kwargs):
+    def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        """Handle GET request to load a transfer."""
         if self.in_progress_submission:
-            self.load_transfer_data(self.in_progress_submission)
+            self.load_form_data()
             return self.render(self.get_form())
 
-        return super().get(self, request, *args, **kwargs)
+        return super().get(request, *args, **kwargs)
+
+    def load_form_data(self) -> None:
+        """Load form data from an InProgressSubmission instance."""
+        if not self.in_progress_submission:
+            raise ValueError("No in-progress submission to load")
+
+        self.storage.data = pickle.loads(self.in_progress_submission.step_data)["past"]
+        self.storage.current_step = self.in_progress_submission.current_step
 
     def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         """Handle POST request to save a transfer."""
         # User is not saving the form, so continue with the normal form submission
         if not request.POST.get("save_form_step", None):
             return super().post(request, *args, **kwargs)
 
-        past_data = self.storage.data
+        try:
+            self.save_form_data(request)
+            messages.success(request, gettext("Transfer saved successfully."))
+        except Exception:
+            messages.error(request, gettext("There was an error saving the transfer."))
+        return redirect("recordtransfer:userprofile")
+
+    def save_form_data(self, request: HttpRequest) -> None:
+        """Save the current state of the form to the database.
+
+        Args:
+            request: The HTTP request object.
+        """
         current_data = TransferFormWizard.format_step_data(self.current_step, request.POST)
+        form_data = {"past": self.storage.data, "current": current_data}
 
         title = None
         if isinstance(current_data, dict):
             title = current_data.get("accession_title")
         if not title:
             title = self.get_form_value(TransferStep.RECORD_DESCRIPTION, "accession_title")
 
-        data = {
-            "save_form_step": self.current_step,
-            "form_data": {"past": past_data, "current": current_data},
-            "submission": self.in_progress_submission,
-            "title": title,
-        }
+        submission = self.in_progress_submission
+        if submission:
+            submission.last_updated = timezone.now()
+        else:
+            submission = InProgressSubmission()
```

Comment:
You can use `self.in_progress_submission` without having to assign it to a different variable:

```suggestion
        if self.in_progress_submission:
            self.in_progress_submission.last_updated = timezone.now()
        else:
            self.in_progress_submission = InProgressSubmission()
```

Reply:
Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/387/commits/1bf338f28ebfd80e9ea7d1e94dd778837581eed6